cosign/README.md

```diff
@@ -0,0 +1,79 @@
+
+## Concepts
+
+Artifact Signatures  
+
+- Artifacts may be signed using different cryptographic schemes. The verifier must support a variety of signature formats, including RSA, ECDSA, and Ed25519.
+- The use of DSSE helps bundle signatures with metadata, providing additional context and validation capabilities.
+
+Transparency Logs  
+
+- Transparency logs, such as those maintained by the Sigstore project, provide an immutable record of signature entries.
+- Verifying against these logs adds an extra layer of assurance that a signature has been publicly disclosed and has not been secretly revoked or altered.
+- As a primary and recommanded solution, Cosign will then store the signature and certificate in the Rekor transparency log.
+
+Verification Workflows  
+
+- The verifier must account for different workflows, such as verifying container images, software binaries, or other artifacts.
+- Each workflow may involve different states of metadata, such as embedded signatures versus detached signatures, and requires customization accordingly.
+- `sigstore-go` library includes a few abstractions to support different use cases, testing, and extensibility:
+
+  - `SignedEntity` - an interface type respresenting a signed message or attestation, with a signature and metadata, implemented by `Bundle`, a type which wraps the `Bundle` type from `protobuf-specs`.
+  - `TrustedMaterial` - an interface type representing a trusted set of keys or certificates for verifying certificates, timestamps, and artifact transparency logs, implemented by `TrustedRoot`
+
+## Scenarios
+
+`sigstore-go` supports multiple verification scenarios based on different signing methods, artifact types, and trust sources.
+These scenarios can be categorized into following main types and for this library we are going to implement the first four verification for basic user scenarios.
+
+| **Verification Scenario**         | **Purpose**                                                            | **Use Case**                                                                 |
+|-----------------------------------|------------------------------------------------------------------------|-----------------------------------------------------------------------------|
+| **Keyless Verification**          | Verifies signature using **OIDC/Fulcio**-based trust (no private key needed). | CI/CD pipelines or automated workflows where no private key management is needed. |
+| **Rekor Transparency Log (TLog)** | Verifies inclusion of the signature in the **Rekor Transparency Log** for audibility. | Auditing and compliance to ensure signatures are publicly recorded in an immutable log. |
+| **Timestamp Verification**        | Verifies the **timestamp** of the signature to prevent time-based attacks. | Long-term signature validity checks, ensuring signatures are valid at a specific point in time. |
```

Comment:
This verification is not standalone, right? It must be combined with something.

Reply:
Updated.